packages/smooth_app/lib/pages/product_list_user_dialog_helper.dart

```diff
@@ -200,99 +200,47 @@ class ProductListUserDialogHelper {
       return false;
     }
 
-    List<String>? selectedLists = <String>[];
-
-    // We only check if the barcode is in the list if we only pass a single barcode
-    if (barcodes.length == 1) {
-      selectedLists = await LoadingDialog.run<List<String>>(
-        context: context,
-        future: daoProductList.getUserLists(withBarcode: barcodes.first),
-      );
-    }
+    final List<String> selectedLists = await LoadingDialog.run<List<String>>(
```

Comment:
I don't think a `LoadingDialog` is appropriate here as we're talking about "instant" database results (though `async`).
If you're really afraid that it may take too long - or if you actually noticed it - that would make sense to embed this computation into the next `LoadingDialog`.
The way it's coded now the user will see stuttering `LoadingDialog`s.

Reply:
You are right it's probably not needed here